[Bug][Failing Test]: LoRA 2 - lora/test_lora_functions.py::test_lora_functions_sync

### Your current environment

N/A

### 🐛 Describe the bug

https://buildkite.com/vllm/ci/builds/20460/steps?jid=0196f343-0fdb-4d91-80da-728e0fb8174c

Summary:
```
[2025-05-21T16:00:09Z] FAILED lora/test_lora_functions.py::test_lora_functions_sync[True] - Exception: Call to add_lora method failed: CUDA error: an illegal memory access was encountered
[2025-05-21T16:00:09Z] CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
[2025-05-21T16:00:09Z] For debugging consider passing CUDA_LAUNCH_BLOCKING=1
[2025-05-21T16:00:09Z] Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.
```

Stack:
```
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559] Invocation of add_lora method failed
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559] Traceback (most recent call last):
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]   File "/usr/local/lib/python3.12/dist-packages/vllm/v1/engine/core.py", line 556, in _handle_client_request
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]     output.result = method(
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]                     ^^^^^^^
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]   File "/usr/local/lib/python3.12/dist-packages/vllm/v1/engine/core.py", line 314, in add_lora
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]     return self.model_executor.add_lora(lora_request)
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]   File "/usr/local/lib/python3.12/dist-packages/vllm/executor/executor_base.py", line 150, in add_lora
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]     return all(self.collective_rpc("add_lora", args=(lora_request, )))
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]   File "/usr/local/lib/python3.12/dist-packages/vllm/executor/uniproc_executor.py", line 56, in collective_rpc
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]     answer = run_method(self.driver_worker, method, args, kwargs)
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]   File "/usr/local/lib/python3.12/dist-packages/vllm/utils.py", line 2598, in run_method
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]     return func(*args, **kwargs)
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]            ^^^^^^^^^^^^^^^^^^^^^
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]   File "/usr/local/lib/python3.12/dist-packages/vllm/v1/worker/gpu_worker.py", line 300, in add_lora
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]     return self.model_runner.add_lora(lora_request)
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]   File "/usr/local/lib/python3.12/dist-packages/vllm/v1/worker/lora_model_runner_mixin.py", line 130, in add_lora
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]     return self.lora_manager.add_adapter(lora_request)
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]   File "/usr/local/lib/python3.12/dist-packages/vllm/lora/worker_manager.py", line 235, in add_adapter
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]     lora = self._load_adapter(lora_request)
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]   File "/usr/local/lib/python3.12/dist-packages/vllm/lora/worker_manager.py", line 141, in _load_adapter
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]     raise e
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]   File "/usr/local/lib/python3.12/dist-packages/vllm/lora/worker_manager.py", line 117, in _load_adapter
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]     lora = self._lora_model_cls.from_local_checkpoint(
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]   File "/usr/local/lib/python3.12/dist-packages/vllm/lora/models.py", line 290, in from_local_checkpoint
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]     return cls.from_lora_tensors(
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]            ^^^^^^^^^^^^^^^^^^^^^^
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]   File "/usr/local/lib/python3.12/dist-packages/vllm/lora/models.py", line 145, in from_lora_tensors
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]     lora_embeddings_tensor.pin_memory())
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559]     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559] RuntimeError: CUDA error: an illegal memory access was encountered
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559] CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559] For debugging consider passing CUDA_LAUNCH_BLOCKING=1
[2025-05-21T15:50:19Z] ERROR 05-21 08:50:19 [core.py:559] Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.
```

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

--- a/tests/lora/test_lora_functions.py
+++ b/tests/lora/test_lora_functions.py
@@ -69,7 +69,7 @@ def test_lora_functions_sync():
     run_check(llm.add_lora, make_lora_request(12), [12, 9, 10, 11])
     run_check(llm.add_lora, make_lora_request(13), [12, 13, 10, 11])
 
-    # Remove all LoRAs
+    # Remove all LoRAs.
     run_check(llm.remove_lora, 13, [12, 10, 11])
     run_check(llm.remove_lora, 12, [10, 11])
     run_check(llm.remove_lora, 11, [10])

--- a/tests/v1/sample/test_topk_topp_sampler.py
+++ b/tests/v1/sample/test_topk_topp_sampler.py
@@ -16,31 +16,40 @@ VOCAB_SIZE = 128 * 1024
 FLASHINFER_ENABLED = current_platform.is_cuda() and is_flashinfer_available
 
 
+@pytest.fixture(autouse=True)
+def reset_default_device():
+    """
+    Explicitly set the default device, which can affect subsequent tests. 
+    Adding this fixture helps avoid this problem.
+    """
+    original_device = torch.get_default_device()
+    yield
+    torch.set_default_device(original_device)
+
+
 def test_topk_impl_equivalance():
 
-    with torch.device(DEVICE):
-        generator = Generator(device=DEVICE).manual_seed(33)
+    torch.set_default_device(DEVICE)
+    generator = Generator(device=DEVICE).manual_seed(33)
 
-        logits = torch.rand((BATCH_SIZE, VOCAB_SIZE), generator=generator)
+    logits = torch.rand((BATCH_SIZE, VOCAB_SIZE), generator=generator)
 
-        # Random top-k values between 1 and 9.
-        k = torch.randint(1, 10, (BATCH_SIZE, ), generator=generator)
+    # Random top-k values between 1 and 9.
+    k = torch.randint(1, 10, (BATCH_SIZE, ), generator=generator)
 
-        # Set k=vocab_size for ~50% of requests in the batch (top-k disabled).
-        k.masked_fill_(
-            torch.randint(0,
-                          2, (BATCH_SIZE, ),
-                          generator=generator,
-                          dtype=bool), VOCAB_SIZE)
+    # Set k=vocab_size for ~50% of requests in the batch (top-k disabled).
+    k.masked_fill_(
+        torch.randint(0, 2, (BATCH_SIZE, ), generator=generator, dtype=bool),
+        VOCAB_SIZE)
 
-        # Top-k only implementation
-        result1 = apply_top_k_top_p(logits=logits.clone(), k=k, p=None)
+    # Top-k only implementation
+    result1 = apply_top_k_top_p(logits=logits.clone(), k=k, p=None)
 
-        # Top-p + top-k
-        no_op_top_p = torch.tensor([1.0])
-        result2 = apply_top_k_top_p(logits=logits.clone(), k=k, p=no_op_top_p)
+    # Top-p + top-k
+    no_op_top_p = torch.tensor([1.0])
+    result2 = apply_top_k_top_p(logits=logits.clone(), k=k, p=no_op_top_p)
 
-        assert torch.allclose(result1, result2)
+    assert torch.allclose(result1, result2)
 
 
 def test_flashinfer_sampler():
@@ -58,50 +67,49 @@ def test_flashinfer_sampler():
         pytest.skip(
             "FlashInfer not installed or not available on this platform.")
 
-    with torch.device(DEVICE):
-        generator = Generator(device=DEVICE).manual_seed(42)
+    torch.set_default_device(DEVICE)
+    generator = Generator(device=DEVICE).manual_seed(42)
 
-        # Generate random logits
-        logits = torch.rand((BATCH_SIZE, VOCAB_SIZE), generator=generator)
+    # Generate random logits
+    logits = torch.rand((BATCH_SIZE, VOCAB_SIZE), generator=generator)
 
-        # Generate various top-k and top-p values
-        k_values = torch.randint(1, 1000, (BATCH_SIZE, ), generator=generator)
-        p_values = torch.rand(
-            (BATCH_SIZE, ),
-            generator=generator) * 0.5 + 0.5  # range in [0.5, 1.0]
+    # Generate various top-k and top-p values
+    k_values = torch.randint(1, 1000, (BATCH_SIZE, ), generator=generator)
+    p_values = torch.rand(
+        (BATCH_SIZE, ), generator=generator) * 0.5 + 0.5  # range in [0.5, 1.0]
 
-        # Sometimes disable top-k (k=vocab_size)
-        k_values.masked_fill_(
-            torch.randint(0,
-                          2, (BATCH_SIZE, ),
-                          generator=generator,
-                          dtype=torch.bool), VOCAB_SIZE)
+    # Sometimes disable top-k (k=vocab_size)
+    k_values.masked_fill_(
+        torch.randint(0,
+                      2, (BATCH_SIZE, ),
+                      generator=generator,
+                      dtype=torch.bool), VOCAB_SIZE)
 
-        # Sometimes disable top-p (p=1.0)
-        p_values.masked_fill_(
-            torch.randint(0,
-                          2, (BATCH_SIZE, ),
-                          generator=generator,
-                          dtype=torch.bool), 1.0)
+    # Sometimes disable top-p (p=1.0)
+    p_values.masked_fill_(
+        torch.randint(0,
+                      2, (BATCH_SIZE, ),
+                      generator=generator,
+                      dtype=torch.bool), 1.0)
 
-        python_logits = apply_top_k_top_p(
-            logits=logits.clone(),
-            k=k_values,
-            p=p_values,
-        )
-        python_probs = torch.softmax(python_logits, dim=-1)
+    python_logits = apply_top_k_top_p(
+        logits=logits.clone(),
+        k=k_values,
+        p=p_values,
+    )
+    python_probs = torch.softmax(python_logits, dim=-1)
 
-        # FlashInfer only exposed renorm interfaces for probs so convert first
-        flashinfer_probs = torch.softmax(logits.clone(), dim=-1)
-        flashinfer_probs = top_k_renorm_probs(
-            probs=flashinfer_probs,
-            top_k=k_values,
-        )
-        flashinfer_probs = top_p_renorm_probs(
-            probs=flashinfer_probs,
-            top_p=p_values,
-        )
+    # FlashInfer only exposed renorm interfaces for probs so convert first
+    flashinfer_probs = torch.softmax(logits.clone(), dim=-1)
+    flashinfer_probs = top_k_renorm_probs(
+        probs=flashinfer_probs,
+        top_k=k_values,
+    )
+    flashinfer_probs = top_p_renorm_probs(
+        probs=flashinfer_probs,
+        top_p=p_values,
+    )
 
-        # Compare the results
-        assert torch.allclose(python_probs, flashinfer_probs, atol=2e-2), \
-            "FlashInfer and Python sampling implementations do not match!"
+    # Compare the results
+    assert torch.allclose(python_probs, flashinfer_probs, atol=2e-2), \
+        "FlashInfer and Python sampling implementations do not match!"

